test/libsolidity/syntaxTests/parsing/errorRecovery/semicolon_error_to_EOS.sol

```diff
@@ -0,0 +1,18 @@
+pragma solidity >=0.0.0;
+
+// This is a contract with a missing semicolon for which there is
+// recovery semicolon to synchronize up to.
+
+// Tokens are deleted until the end of stream.
+
+contract ErrorStatementToEOS {
+  function five() returns(int) {
+    uint256 a;
+    a = 1
+    b = 2
+  }
+}
+// ----
+// ParserError: (255-256): Expected ';' but got identifier. Cannot find ';' to synchronize to.
```

Comment:
I think "Skipping to next X" is a message people can understand and might find useful yet not too confusing. On the other hand "Cannot find ';' to synchronize to" is probably confusing. It is an error message inside an error message :)
We can either leave it at a fatal error (and potentially synchronize one level higher in the try-catch handler) or just not report that fact.

Reply:
Ah, I just saw that this is only used when we reach end-of-source. In that case, the user-friendliest approach would be to not emit any error message anymore, but this is probably complicated to implement and might cause other side-effects.

Reply:
> We can either leave it at a fatal error (and potentially synchronize one level higher in the try-catch handler) or just not report that fact.

Thanks for the suggestion. I will look into resetting the input stream pointer back and issuing a fatal error for a higher level to try synchronization.

Reply:
@chriseth I haven't gotten around to this yet, but will sometime this week. 

Looking at error-recovery results I probably will add try/catch recovery to `parseBlock` as expermients suggest that can help a lot.

Reply:
If we run off the end now, we don't give a Warning message to that effect.